src/Parallel/TypeTraits.hpp

```diff
@@ -0,0 +1,33 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines type traits related to Charm++ types
+
+#pragma once
+
+#include <charm++.h>
+
+namespace Parallel {
+
+/// \ingroup Parallel
```

Comment:
I thought about putting these in Utilities/TypeTraits.hpp and decided they were too tied to Charm++;  a different question is should the PUP stuff be moved out of Utilities/TypeTraits and into Parallel/TypeTraits...